libvast/src/node.cpp

```diff
@@ -547,34 +548,25 @@ node(node_actor::stateful_pointer<node_state> self, std::string /*name*/,
       return run(inv, self->system(), node_state::command_factory);
     },
     [self](atom::proxy,
-           http_request_description& desc) -> caf::result<std::string> {
+           http_request_description& desc) -> caf::result<rest_response> {
       VAST_VERBOSE("{} proxying request to {}", *self, desc.canonical_path);
       auto [handler, endpoint] = self->state.get_endpoint_handler(desc);
       if (!handler)
         return caf::make_error(ec::system_error,
                                "failed to spawn rest handler");
-      auto rp = self->make_response_promise<std::string>();
-      auto response = std::make_shared<detail::internal_http_response>(rp);
+      auto rp = self->make_response_promise<rest_response>();
       auto params = parse_endpoint_parameters(endpoint, desc.params);
       if (!params)
         return caf::make_error(ec::invalid_argument, "invalid parameters");
```

Comment:
I believe we should return a `rest_response` here and on line 555.

Reply:
Oh, and you should wait to create the response promise until all the errors are checked.